core/src/main/java/com/novoda/noplayer/drm/LocalDrmSession.java

```diff
@@ -0,0 +1,104 @@
+package com.novoda.noplayer.drm;
+
+import android.annotation.TargetApi;
+import android.media.MediaCryptoException;
+import android.media.MediaDrmException;
+import android.os.Build;
+import android.os.Looper;
+
+import com.google.android.exoplayer2.drm.DrmInitData;
+import com.google.android.exoplayer2.drm.DrmSession;
+import com.google.android.exoplayer2.drm.DrmSessionManager;
+import com.google.android.exoplayer2.drm.ExoMediaDrm;
+import com.google.android.exoplayer2.drm.FrameworkMediaCrypto;
+import com.google.android.exoplayer2.drm.FrameworkMediaDrm;
+import com.google.android.exoplayer2.drm.UnsupportedDrmException;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.UUID;
+
+@TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
+class LocalDrmSession implements DrmSessionManager<FrameworkMediaCrypto>, DrmSession<FrameworkMediaCrypto> {
```

Comment:
I'm confused because it seems like this should be two different objects. Should we not have a manager, responsible for acquiring a session, and a `DrmSession`, from which you can retrieve things like `FrameworkMediaCrypto`. Why were these originally coupled together?